tools/analyzer_plugin/tool/doc.dart

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2015, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:args/args.dart';
+import 'package:glob/glob.dart';
+import 'package:over_react_analyzer_plugin/src/doc_utils/contributor_meta_registry.dart';
+import 'package:over_react_analyzer_plugin/src/doc_utils/documented_contributor_meta.dart';
+import 'package:over_react_analyzer_plugin/src/util/constants.dart';
+
+import 'doc_generation_utils/config.dart';
+import 'doc_generation_utils/generate_assist_docs.dart';
+import 'doc_generation_utils/generate_rule_docs.dart';
+import 'doc_generation_utils/visitor.dart';
+
+final configs = [
+  DocsGenerationConfig(
+    srcDir: 'diagnostic',
+    outputSubDir: analyzerPluginLintDocsDir,
+    registry: DiagnosticMetaRegistry.diagnosticRegistry,
+    getSortedContributorMetas: () => RulesIndexer.rules,
+    typeNameOfContributorClass: 'DiagnosticContributor',
+    typeNameOfAnnotatedField: 'DiagnosticCode',
+    getMeta: (el) => DocumentedDiagnosticContributorMeta.fromAnnotatedFieldAst(el),
+    // For now, we'll ignore the implicit cast issue here since attempts to parameterize the config result
+    // in runtime exceptions of the form:
+    //
+    //     type '(DocumentedDiagnosticContributorMeta) => RuleDocGenerator' is not a subtype of type '(DocumentedContributorMetaBase) => ContributorDocPage<DocumentedContributorMetaBase>'
+    //
+    // ignore: argument_type_not_assignable
+    getPageGenerator: (meta) => RuleDocGenerator(meta),
```

Comment:
This just needs a cast
```suggestion
    getPageGenerator: (meta) => RuleDocGenerator(meta as DocumentedDiagnosticContributorMeta),
```